sdk/monitor/query/MIGRATION.md

```diff
@@ -0,0 +1,211 @@
+    res, err := client.QueryResource(context.TODO(), "<resourceID>",
+        &azquery.MetricsClientQueryResourceOptions{
+            Timespan:        to.Ptr(azquery.NewTimeInterval(time.Date(2022, 12, 25, 0, 0, 0, 0, time.UTC), time.Date(2022, 12, 25, 12, 0, 0, 0, time.UTC))),
+            Interval:        to.Ptr("PT1M"),
+            MetricNames:     nil,
+            Aggregation:     to.SliceOfPtrs(azquery.AggregationTypeAverage, azquery.AggregationTypeCount),
+            Top:             to.Ptr[int32](3),
+            OrderBy:         to.Ptr("Average asc"),
+            Filter:          to.Ptr("BlobType eq '*'"),
+            ResultType:      nil,
+            MetricNamespace: to.Ptr("Microsoft.Storage/storageAccounts/blobServices"),
+        })
+    if err != nil {
+        //TODO: handle error
+    }
+}
+```
+
+### `armmonitor`
+
+The code in `armmonitor` is closer to the REST API; therefore, there are some name changes. Additionally, `azquery.MetricsClient` is separated into [different clients](#name-changes).
+
+```go
+import (
+    "context"
+	"log"
+
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/monitor/armmonitor"
+)
+
+func main() {
+    // create the client
+    cred, err := azidentity.NewDefaultAzureCredential(nil)
+    if err != nil {
+        //TODO: handle error
+    }
+    client, err := armmonitor.NewMetricsClient("<subscription-id>", cred, nil)
+    if err != nil {
+        //TODO: handle error
+    }
+
+    // execute the query
+    res, err := client.List(context.Background(), "<resourceID>", &armmonitor.MetricsClientListOptions{
+        Timespan:            to.Ptr("2021-04-20T09:00:00.000Z/2021-04-20T14:00:00.000Z"),
+        Interval:            to.Ptr("PT6H"),
+        Metricnames:         to.Ptr("BlobCount,BlobCapacity"),
+        Aggregation:         to.Ptr("average,minimum,maximum"),
+        Top:                 to.Ptr[int32](5),
+        Orderby:             to.Ptr("average asc"),
+        Filter:              to.Ptr("Tier eq '*'"),
+        ResultType:          nil,
+        Metricnamespace:     to.Ptr("Microsoft.Storage/storageAccounts/blobServices"),
+        AutoAdjustTimegrain: to.Ptr(true),
+        ValidateDimensions:  to.Ptr(false),
+    })
+    if err != nil {
+        //TODO: handle partial error
+    }
+}
+```
+
+<!-- LINKS -->
+[armmonitor]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/monitor/armmonitor
+[azlogs]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/monitor/query/azlogs
+[azmetrics]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/monitor/query/azmetrics
+[azmetrics_blog]: https://devblogs.microsoft.com/azure-sdk/multi-resource-metrics-query-support-in-the-azure-monitor-query-libraries/
+[github_issues]: https://github.com/Azure/azure-sdk-for-go/issues
+[rest_api]: https://learn.microsoft.com/en-us/rest/api/monitor/
```

Comment:
Omit the locale string

Reply:
fixed with [c79390d](https://github.com/Azure/azure-sdk-for-go/pull/22753/commits/c79390ddba46a599c6717d0bde75b9d3ccb46996)